### PR TITLE
[SEMI-MODULAR] SR specific escape menu options

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request
Modifies the new escape menu to fit SR.

- Removes suicide
- Adds OPFOR
- Adds ghost on main menu
- Adds respawn

## How This Contributes To The Skyrat Roleplay Experience
Puts those features in a more accessible place.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image

</details>

## Changelog

:cl: LT3
code: Escape menu now has SR specific line items
/:cl:

### CODE REVIEW

**Comment by tf-4** [comment]
IMO this is a feature, so it falls under the freeze

**Comment by lessthnthree** [comment]
Okay, I'll just reopen it later.

**Comment by tf-4** [comment]
I would be happy with an atomised removal of the suicide stuff tho

**Comment by lessthnthree** [comment]
> I would be happy with an atomised removal of the suicide stuff tho

I already removed suicide first merge, so any changes to the menu we can roll into one after freeze

File: code/modules/escape_menu/escape_menu.dm [hunk 1]
```diff
@@ -80,8 +80,8 @@ GLOBAL_LIST_EMPTY(escape_menus)
 	switch (menu_page)
 		if (PAGE_HOME)
 			show_home_page()
-		if (PAGE_LEAVE_BODY)
-			show_leave_body_page()
+//		if (PAGE_LEAVE_BODY) //SKYRAT EDIT REMOVAL
+//			show_leave_body_page() //SKYRAT EDIT REMOVAL
```
Comment (tf-4) [inline code/modules/escape_menu/escape_menu.dm]:
```suggestion
		//if (PAGE_LEAVE_BODY) //SKYRAT EDIT REMOVAL
			//show_leave_body_page() //SKYRAT EDIT REMOVAL
```

File: code/modules/escape_menu/escape_menu.dm [hunk 2]
```diff
@@ -116,6 +116,12 @@ GLOBAL_LIST_EMPTY(escape_menus)
 	plane_master_controller = plane_master_controllers[PLANE_MASTERS_NON_MASTER]
 	plane_master_controller.add_filter("escape_menu_blur", 1, list("type" = "blur", "size" = 2))
 
+/datum/escape_menu/proc/respawn()
+	PRIVATE_PROC(TRUE)
+
+	var/mob/living/client_mob = client?.mob
+	client_mob?.abandon_mob()
+
```
Comment (tf-4) [inline code/modules/escape_menu/escape_menu.dm]: modular file

File: code/modules/escape_menu/home_page.dm [hunk 3]
```diff
@@ -28,23 +28,69 @@
 		)
 	)
 
+// SKYRAT EDIT ADDITION START
```
Comment (tf-4) [inline code/modules/escape_menu/home_page.dm]: this isn't purely an addition

File: code/modules/escape_menu/home_page.dm [hunk 4]
```diff
@@ -28,23 +28,69 @@
...
+	qdel(src)
+
+/datum/escape_menu/proc/home_opfor()
+	PRIVATE_PROC(TRUE)
+
+	// Not guaranteed to be living. Everything defines verb/ghost separately. Fuck you.
+	var/mob/living/living_user = client?.mob
+	living_user?.opposing_force()
+	qdel(src)
+//SKYRAT EDIT ADDITION END
```
Comment (tf-4) [inline code/modules/escape_menu/home_page.dm]: just put this in a modular file

File: code/modules/escape_menu/home_page.dm [hunk 5]
```diff
@@ -28,23 +28,69 @@
...
+	page_holder.give_screen_object(
+		new /atom/movable/screen/escape_menu/home_button/respawn(
+			null,
+			src,
+			"Respawn",
+			/* offset = */ 5,
+			CALLBACK(src, PROC_REF(home_respawn)),
+		)
+	)
+	//SKYRAT EDIT ADDITION END
```
Comment (tf-4) [inline code/modules/escape_menu/home_page.dm]: should be able to modularise this

File: code/modules/escape_menu/home_page.dm [hunk 6]
```diff
@@ -283,3 +329,39 @@
...
+	on_click_callback,
+)
+	. = ..()
+
+/atom/movable/screen/escape_menu/home_button/opfor/enabled()
+	if (!..())
+		return FALSE
+
+	return isliving(escape_menu.client?.mob)
+// SKYRAT ADDITION END
```
Comment (tf-4) [inline code/modules/escape_menu/home_page.dm]: modularrrrrr

File: code/modules/escape_menu/leave_body.dm [hunk 7]
```diff
@@ -1,134 +0,0 @@
-/datum/escape_menu/proc/show_leave_body_page()
```
Comment (tf-4) [inline code/modules/escape_menu/leave_body.dm]: dont delete this file

**Review by tf-4 — CHANGES_REQUESTED** [review]

File: code/modules/escape_menu/leave_body.dm (hunk 7 above)
Comment (Gandalf2k15) [reply code/modules/escape_menu/leave_body.dm]: ^

File: tgstation.dme [hunk 8]
```diff
@@ -3209,7 +3209,6 @@
 #include "code\modules\escape_menu\dimmer.dm"
 #include "code\modules\escape_menu\escape_menu.dm"
 #include "code\modules\escape_menu\home_page.dm"
-#include "code\modules\escape_menu\leave_body.dm"
```
Comment (Gandalf2k15) [inline tgstation.dme]: make sure to tick this file

Comment (lessthnthree) [reply tgstation.dme]: Done

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**